features/all-positions/PositionActionsDialog.tsx

```diff
@@ -103,6 +104,40 @@ const PositionActionsDialog = (props: DialogProps) => {
   const [success, setSuccess] = useState<boolean | null>(null);
   const [error, setError] = useState<Error | null>(null);
 
+  useEffect(() => {
+    if (
+      props.selectedSponsor !== null &&
+      latestPrice !== null &&
+      collReq !== null
+    ) {
+      const sponsorPosition = activeSponsors[props.selectedSponsor];
+
+      // Set smart min and max collPerToken params for the Liquidation.
+      // - min collPerToken should be the point at which you'll lose money on the liquidation
+      //   (i.e. when locked collateral in the position is = 100% of the token debt).
+      //   Since the liquidator must "burn" tokens to liquidate and withdraw collateral,
+      //   the amount of backing collateral in the position must be >= 100% of the token debt.
+      //   Therefore, the minimum `collPerToken` is simply the current price of 1 unit token debt,
+      //   which is the `latestPrice`.
+      // - max collPerToken should be point at which the liquidation is invalid
+      //   (i.e. equal to the coll requirement).
+      //   The maximum amount of backing collateral in the position for it to be liquidatable
+      //   is 1.25 x the current price of the token debt, which simplifies to 1.25 `latestPrice`.
+      // - (TODO) If the current ratio of backing collateral to token debt is < `minCollPerToken`
+      //   or > `maxCollPerToken`, then the user shouldn't be liquidating the position.
+      //   Perhaps we should block this functionality in this situation.
```

Comment:
Yeah, I was thinking about this too. It seems weird to set the min above where the position currently is or set the max below where it currently is since the txn would be expected to revert.

Agree with you 100%. Maybe (?) it makes sense to:
1. Block if the field contains a value that would cause the transaction to revert.
2. Warn them if they populate the field with a value that will allow the transaction to go through, but we think that they will lose money on the liquidation (either because it will be invalid or there's not enough collateral).

But ^ is just a straw man.

Reply:
yeah I am leaning more towards 2 but also we should disable the button if the transaction would revert.

Reply:
I want to leave this for another PR as this likely calls for a signficiant UI overhaul in the `PositionDialog` feature

https://github.com/UMAprotocol/emp-tools/issues/129